src/fullerite/collector/docker_stats.go

```
@@ -308,3 +387,140 @@ func min(x, y int) int {
 	}
 	return y
 }
+
+// returns a map, each record containing (device name --> [disk stats])
+func ObtainDiskStats(fn string) (map[string][]string, error) {
+    devNameMinMajMap := make(map[string][]string)
+    var major int
+
+    file, err := os.Open(fn)
+    defer file.Close()
+
+    if err != nil {
+        return devNameMinMajMap, err
+    }
+
+    // read the contents of the file with a reader.
+    reader := bufio.NewReader(file)
+
+    // read line-by-line
+    var line string
+    for {
+        line, err = reader.ReadString('\n')
+
+        if err != nil {
+            break
+        }
+
+        // split the line on space
+       rec := strings.Fields(line)
+       // filter out system devices and any other which are not EC2 (major != 202). For reference, https://cromwell-intl.com/cybersecurity/ec2-secure-storage.html
+       major, err = strconv.Atoi(rec[0])
+       if err != nil {
+           continue
```

Why ignoring errors here?